flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveTableSource.java

```diff
@@ -247,6 +254,30 @@ public void applyPartitions(List<Map<String, String>> remainingPartitions) {
         }
     }
 
+    @Override
+    public List<String> applyDynamicFiltering(List<String> candidateFilterFields) {
+        if (catalogTable.getPartitionKeys() != null
+                && catalogTable.getPartitionKeys().size() != 0) {
+            checkArgument(
+                    !candidateFilterFields.isEmpty(),
+                    "At least one field should be provided for dynamic filtering");
+            checkState(
+                    dynamicPartitionKeys == null, "Dynamic filtering should not be applied twice.");
+
+            // only accept partition fields to do dynamic partition pruning
+            this.dynamicPartitionKeys = new ArrayList<>();
+            for (String field : candidateFilterFields) {
+                if (catalogTable.getPartitionKeys().contains(field)) {
```

Comment:
Shoud we warn when meets a field that not exists in `catalogTable.getPartitionKeys`?

Reply:
I  suppose we don't need to. According to the description of the interface SupportsDynamicFiltering, source can pick the fields that they can use for dynamic filtering. It's expected that some fields are not supported here.

Reply:
In my opinion even rejecting all fields is also acceptable, which is the case of non-partition hive tables. The result is still correct, and as the planner should not add dynamic filter data collector in such cases, there should be no impact to performance. So I think it's ok not to do check and let the job run normally. But maybe a warning log can be added here to notify the users about this.

Reply:
Thanks for the remind! I'll remove the throwing but add a warning log instead